app/controllers/local_transaction_controller.rb

```diff
@@ -0,0 +1,132 @@
+class LocalTransactionController < ApplicationController
+  include ActionView::Helpers::TextHelper
+
+  before_filter :validate_slug_param, only: :search
+  before_filter -> { setup_content_item_and_navigation_helpers("/" + params[:slug]) }
+  before_filter -> { response.headers['X-Frame-Options'] = 'DENY' }
+  before_filter -> { set_expiry unless viewing_draft_content? }
+
+  INVALID_POSTCODE = 'invalidPostcodeFormat'
+  NO_AUTHORITY_URL = 'laMatchNoLinkNoAuthorityUrl'
+  NO_LINK = 'laMatchNoLink'
+  NO_MAPIT_MATCH = 'fullPostcodeNoMapitMatch'
+  NO_MATCHING_AUTHORITY = 'noLaMatch'
+
+  def search
+    @publication = publication
+    @edition = params[:edition]
+
+    if request.post?
+      @postcode = postcode
+      postcode_search; return if performed?
+    end
+
+    respond_to do |format|
+      format.html { render }
+      format.json { render json: @publication.to_json }
+    end
+  end
+
+  def results
+    @publication = publication
+    @postcode = postcode
+    @edition = params[:edition]
+    @interaction_details = interaction_details
+    @local_authority = local_authority
+  end
+
+private
+
+  def postcode_search
+    if postcode.nil?
+      @location_error = LocationError.new(INVALID_POSTCODE)
+      return
+    end
+
+    if local_authority_slug
+      return redirect_to local_transaction_results_path(local_authority_slug: local_authority_slug)
```

Comment:
I don't think that the return value of `redirect_to` is part of it's public API.  Might it be a little more comfortable to phrase this differently so it always returns truthy, regardless of how `redirect_to` responds?